test/transcoding/SPV_INTEL_function_pointers/fp-in-recusive-type.ll

```diff
@@ -0,0 +1,84 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_function_pointers -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.spv -r -o %t.r.bc
+; RUN: llvm-dis %t.r.bc -o - | FileCheck %s --check-prefix=CHECK-LLVM
+;
+; This test is intended to check the following source code which caused
+; assertion failure and subsequent crash:
+;
+; virtual SPIRV::SPIRVEntry* SPIRV::SPIRVModuleImpl::getEntry(SPIRV::SPIRVId) const: Assertion `Loc != IdEntryMap.end() && "Id is not in map"' failed.
+;
+; typedef void(*fp_t)(struct Desc *);
+;
+; struct Type;
+;
+; struct Desc {
+;   int B;
+;   Type *Ptr;
+; };
+;
+; struct Type {
+;   int A;
+;   fp_t FP;
+; };
+;
+; Type T;
+;
+;
+; CHECK-SPIRV: TypeInt [[INTTY:[0-9]+]] 32 0
+; CHECK-SPIRV: TypeVoid [[VOIDTY:[0-9]+]]
+; CHECK-SPIRV: TypeStruct [[TYPETY:[0-9]+]] [[INTTY]] [[FPTRTY:[0-9]+]]
+; CHECK-SPIRV: TypeStruct [[DESCTY:[0-9]+]] [[INTTY]] [[TYPEPTRTY:[0-9]+]]
+; CHECK-SPIRV: TypePointer [[TYPEPTRTY]] {{[0-9]+}} [[TYPETY]]
+; CHECK-SPIRV: TypePointer [[DESCPTRTY:[0-9]+]] {{[0-9]+}} [[DESCTY]]
+; CHECK-SPIRV: TypeFunction [[FTY:[0-9]+]] [[VOIDTY]] [[DESCPTRTY]]
+; CHECK-SPIRV: TypePointer [[FPTRTY]] {{[0-9]+}} [[FTY]]
+; CHECK-SPIRV: TypePointer [[DESCPTR1TY:[0-9]+]] {{[0-9]+}} [[TYPETY]]
+; CHECK-SPIRV: Variable [[DESCPTR1TY]]
+;
+; CHECK-LLVM: %[[TYPE:.*]].Type = type { i32, void (%[[DESC:.*]].Desc addrspace(4)*) addrspace(4)* }
```

Comment:
Did you want to capture just `struct` or `struct.Type` ?
The same for question for `DESC`.

Reply:
Oh, fair point, there is no sense in capturing `struct` and I don't really need any filecheck variable here. I think I will remove `%[[TYPE]]` and `%[[DESC]]` at all and replace it with `struct.Type` and `struct.Desc`